src/main/java/com/spring/boot/project/demo/dto/review/ReviewMapper.java

```diff
@@ -12,7 +13,8 @@ public ReviewFromDbDto convertToReviewFromDbDto(Review review) {
         reviewFromDbDto.setId(review.getId());
         reviewFromDbDto.setAmazonUserDto(new AmazonUserMapper()
                 .convertToAmazonUserDto(review.getAmazonUser()));
-        reviewFromDbDto.setProductId(review.getProduct().getProductId());
+        reviewFromDbDto
+                .setProductDto(new ProductMapper().convertToAmazonUserDto(review.getProduct()));
```

Comment:
ProductMapper в тебе компонент. Не використовуй new в SPRING! Заюзай autowired
```suggestion
                .setProductDto(productMapper.convertToAmazonUserDto(review.getProduct()));
```